.github/pull_request_template.md

```diff
@@ -1,39 +1,12 @@
-Notes:
-* Pull requests will not be accepted until the submitter has agreed to the [contributer agreement](https://github.com/ARMmbed/mbed-os/blob/master/CONTRIBUTING.md).
-* This is just a template, so feel free to use/remove the unnecessary things
-
-## Description
-A few sentences describing the overall goals of the pull request's commits.
-
-
-## Status
-**READY/IN DEVELOPMENT/HOLD**
-
-
-## Migrations
-If this PR changes any APIs or behaviors, give a short description of what *API users* should do when this PR is merged.
-
-YES | NO
-
-
-## Related PRs
-List related PRs against other branches:
-
-branch | PR
------- | ------
-other_pr_production | [link]()
-other_pr_master | [link]()
-
-
-## Todos
-- [ ] Tests
-- [ ] Documentation
-
-
-## Deploy notes
-Notes regarding the deployment of this PR. These should note any
-required changes in the build environment, tools, compilers, etc.
-
-
-## Steps to test or reproduce
-Outline the steps to test or reproduce the PR here.
+```diff
+- NOTE: Pull requests submitted without the following checklist being completed and evidenced will not be accepted -
+```
+Checklist item is marked as complete by adding a X to the box.
+
+- [ ] [Contributer agreement](https://github.com/ARMmbed/mbed-os/blob/master/CONTRIBUTING.md) has been signed.
+- [ ] A detailed description of the PR purpose along with a meaningful title have been provided.
```

Comment:
For the title requirement, the real concern is that it doesn't contain Markdown syntax, correct? And this stems from the fact that PR titles are used in release notes. If I knew my little PR was going to be included in release notes, I'd probably spend more time on it 😄 For that reason, perhaps the following would serve this purpose better:

```
- [ ] A meaningful title that follows [good commit message structure](https://chris.beams.io/posts/git-commit/)
    - This title is used in the release notes, so please do not use [Markdown syntax](https://guides.github.com/features/mastering-markdown/)
```

 This is what it looks like formatted:

- [ ] A meaningful title that follows [good commit message structure](https://chris.beams.io/posts/git-commit/)
    - This title is used in the release notes, so please do not use [Markdown syntax](https://guides.github.com/features/mastering-markdown/)